app/uk/gov/hmrc/cdsreimbursementclaim/services/DeclarationInfoService.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaim.services
+
+import cats.data.EitherT
+import cats.syntax.eq._
+import javax.inject.{Inject, Singleton}
+import play.api.http.Status
+import play.api.libs.json.JsValue
+import uk.gov.hmrc.cdsreimbursementclaim.connectors.DeclarationInfoConnector
+import uk.gov.hmrc.cdsreimbursementclaim.models.Error
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+@Singleton
+class DeclarationInfoService @Inject()(declarationInfoConnector: DeclarationInfoConnector)(implicit ec: ExecutionContext) extends Logging {
+
+  def getDeclarationInfo(claimData: JsValue)(implicit hc: HeaderCarrier): EitherT[Future, Error, JsValue] =
```

Comment:
rf: function parameter -> this is not `claimData` you are retrieving but rather `declarationData`